code/device/viewer.h

```diff
@@ -13,20 +18,59 @@ enum viewer_error_code
 };
 
 
-class viewer
+class viewer: public QGraphicsView
 {
+    Q_OBJECT
+    enum renderer_type
+    {
+        RENDERER_NATIVE,
+        RENDERER_OPENGL,
+        RENDERER_IMAGE
+    };
+
  public:
-    void InitMap(svg_view* newMapPic);          // Saving new map imame
+    explicit viewer(QWidget *parent = nullptr);
+    bool InitMap(const QString &fileName);          // Saving new map imame
     void ClearMap();                            // deleting img ptr
 
     void ViewMap();                             // Draws map
     void ViewObject();                          // Drows something (preparing for future) f.e. selected shop
     void ViewGraph();                           // Shows graph on map (for debug )
     void ViewPath(coord *from, coord *to);      // Shows path from firt point to second point (calls Graph)
+    qreal zoomFactor();
+    qreal GetMapPicScale();
+    virtual ~viewer() {};
```

Comment:
memory leak at mapScene, must delete it here, and delete all objects created in init with "new"